File: app/src/main/java/com/alphacorporations/givememymoney/model/repositories/DebtRepository.java
package com.alphacorporations.givememymoney.model.repositories;

import androidx.lifecycle.LiveData;

import com.alphacorporations.givememymoney.model.Debt;
import com.alphacorporations.givememymoney.model.database.dao.DebtDao;

import java.util.List;

/**
 * Created by Alph4 le 24/05/2020.
 * Projet: Give Me My Money
 **/
public class DebtRepository {

    private final DebtDao mDebtDao;

    public DebtRepository(DebtDao debtDao) { mDebtDao = debtDao; }

    //--- GET ---
    public LiveData getDebt(long debtId) {
        return this.mDebtDao.getDebt(debtId);
    }

    public LiveData<List<Debt>> getDebts() {
        return this.mDebtDao.getDebts();
    }

    //--- INSERT ---
    public void insertDebt(Debt debt) {
        this.mDebtDao.insertDebt(debt);
    }

    //--- UPDATE ---
    public void updateDebt(Debt debt) {
        this.mDebtDao.updateDebt(debt);
    }

    //--- DELETE ---
    public void deleteDebt(long idDebt) {
        this.mDebtDao.deleteDebt(idDebt);
    }
}

File: app/src/main/java/com/alphacorporations/givememymoney/model/repositories/DebtRepository.kt
package com.alphacorporations.givememymoney.model.repositories

import androidx.lifecycle.LiveData
import com.alphacorporations.givememymoney.model.Debt
import com.alphacorporations.givememymoney.model.database.dao.DebtDao

/**
 * Created by Alph4 le 24/05/2020.
 * Projet: Give Me My Money
 */
class DebtRepository(private val mDebtDao: DebtDao) {

    //--- GET ---
    fun getDebt(debtId: Long): LiveData<Debt?>? {
        return mDebtDao.getDebt(debtId)
    }

    val debts: LiveData<List<Debt?>?>?
        get() = mDebtDao.debts

    //--- INSERT ---
    fun insertDebt(debt: Debt?) {
        mDebtDao.insertDebt(debt)
    }

    //--- UPDATE ---
    fun updateDebt(debt: Debt?) {
        mDebtDao.updateDebt(debt)
    }

    //--- DELETE ---
    fun deleteDebt(idDebt: Long) {
        mDebtDao.deleteDebt(idDebt)
    }

}